MatrixSDK/Data/Store/MXFileStore/MXFileStore.m

```diff
@@ -606,8 +597,35 @@ - (void)commit
     }
 }
 
+- (void)saveDataToFiles
+{
+    // Save each component one by one
+    [self saveRoomsDeletion];
+    [self saveRoomsMessages];
+    [self saveRoomsState];
+    [self saveRoomsSummaries];
+    [self saveRoomsAccountData];
+    [self saveReceipts];
+    [self saveUsers];
+    [self saveGroupsDeletion];
+    [self saveGroups];
+    [self saveMetaData];
+}
+
 - (void)close
 {
+    NSLog(@"[MXFileStore] close: %tu pendingCommits", pendingCommits);
+
+    // Flush pending commits
+    if (pendingCommits)
+    {
+        [self saveDataToFiles];
+
+        dispatch_sync(dispatchQueue, ^(void){
```

Comment:
it should be `dispatch_async`, no?

Reply:
No, we want to synchronously close the store.

Reply:
No, we want to synchronously close the store.

Reply:
Well, sync or async is not really an issue. The method ends with a sync dispatch to ensure every dispatched task is done.
Using sync here will just ensure that all tasks are complete a bit before.

Reply:
ok